src/utils.ts

```diff
@@ -108,3 +111,15 @@ export function encodeUnicode(value: any): string {
   const binary = encodeURIComponent(str).replace(/%([0-9A-F]{2})/g, toSolidBytes);
   return btoa(binary);
 }
+
+export async function retry<T>(action: () => Promise<T>, retries = 2): Promise<T> {
+  for (let i = 0; i <= retries; i++) {
+    try {
+      return await action();
+    } catch (error) {
+      if (i + 1 > retries) {
+        throw error;
+      }
+    }
+  }
+}
```

Comment:
This function is used to automatically retry some upload calls. Those tend to be more susceptible to unexpected network errors since they are passing larger files. In many cases a failing asset creation call is hard to repeat later if we want it in the right spot in a kit.